doc: Add VSC instructions to nRF53 user guide (#7570)
nrfconnect/sdk-nrf: doc/nrf/ug_nrf5340.rst

#. Program the application sample using :guilabel:`Target` > :guilabel:`Download zephyr/merged.hex`.

Using |VSC| extension
======================

You can build and program separate images or combined images using |VSC|.

Separate images
---------------

> To build and program only the application core, follow the instructions in :`Building an application`_ and use ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as build target.

Reviewer: ```suggestion
To build and program only the application core, follow the instructions in :`Building an application`_ and use ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as the build target.
```

> To build and program only the application core, follow the instructions in :`Building an application`_ and use ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as the build target.

Reviewer 2: ```suggestion
To build and program only the application core, follow the instructions in `Building an application`_ and use ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as the build target.
```

> To build and program a dedicated network sample, follow the instructions in `Building an application`_ and use ``nrf5340dk_nrf5340_cpunet`` as the build target.

Author: What I understood from here is : http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf5340.html#id18, this would be just for the network core.
I have rephrased it a bit

.. _ug_nrf5340_VSC_multi_image:

Multi-image build
-----------------

> If you are working with Bluetooth LE, Thread, Zigbee, or Matter samples, the network core sample is built as child image when you build the application core image (see :ref:`ug_nrf5340_multi_image` above).

Reviewer: ```suggestion
If you are working with Bluetooth LE, Thread, Zigbee, or Matter samples, the network core sample is built as a child image when you build the application core image (see :ref:`ug_nrf5340_multi_image` above).
```

Reviewer: I think this is still missing the article `a`?

Author: Its there :)

Reviewer 3: I am also not seeing the "a" there. I have just refreshed my browser several times but it is still missing. Maybe GitHub is not showing the changes correctly, but can you double check just in case? I noticed there was a similar change on line 372, so can you confirm you are also seeing it on line 518 as well?

![Screenshot 2022-05-18 143158](https://user-images.githubusercontent.com/17974570/169039343-e60706c0-9166-4096-8180-7154c3c50099.png)
.

> Follow these steps to build and program a multi-image build to the nRF5340 application core and network core:

Reviewer 4: ```suggestion
Complete the following steps to build and program a multi-image build to the nRF5340 application core and network core:
```

.. include:: /includes/vsc_build_and_run_nrf53.txt

6. Select the ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as the target board and ``prj.conf`` as the configuration.
#. Select the :guilabel:`Build after generating configuration` checkbox and click on the :guilabel:`Build Configuration` button.
   This generates the configuration file and triggers the build process.

> #. When the build configuration and the build are complete, an :guilabel:`Actions` window appears.

Reviewer: ```suggestion
#. When the build configuration and the build are complete, an :guilabel:`Actions` panel appears.
```
Maybe this is fine at is, but in the VSC docs we refer to these sections as `panels`.

>    In this window, you can trigger the build process, program the built sample, or start a debug session.

Reviewer: ```suggestion
   In this panel, you can trigger the build process, program the built sample, or start a debug session.
```
Please see previous comment.

#. Program the application sample using :guilabel:`Target` > :guilabel:`Download zephyr/merged.hex`.

> Using |VSC| extension

Reviewer 3: ```suggestion
Using |VSC|
```

===========

|vsc_extension_instructions|

You can build and program separate images or combined images using |VSC|.

Separate images
---------------

To build and program the application core, follow the instructions in `Building an application`_ and use ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as the build target.

To build and program the network core, follow the instructions in `Building an application`_ and use ``nrf5340dk_nrf5340_cpunet`` as the build target.

.. _ug_nrf5340_VSC_multi_image:

Multi-image build
-----------------

If you are working with Bluetooth LE, Thread, Zigbee, or Matter samples, the network core sample is built as a child image when you build the application core image (see :ref:`ug_nrf5340_multi_image` above).

Complete the following steps to build and program a multi-image build to the nRF5340 application core and network core:

.. include:: /includes/vsc_build_and_run_nrf53.txt

> 6. Select the ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as the target board and ``prj.conf`` as the configuration.

Reviewer 5: ```suggestion
6. Select the ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as the target board.
```

Whether you use prj.conf or not is sample specific. The extension will select an appropriate conf file by default.

Separate images
---------------

To build and program only the application core, follow the instructions in :`Building an application`_ and use ``nrf5340dk_nrf5340_cpuapp`` or ``nrf5340dk_nrf5340_cpuapp_ns`` as build target.

To build and program a dedicated network sample, follow the instructions in `Building an application`_ and use ``nrf5340dk_nrf5340_cpunet`` as the build target.

.. _ug_nrf5340_VSC_multi_image:

Multi-image build
-----------------

If you are working with Bluetooth LE, Thread, Zigbee, or Matter samples, the network core sample is built as child image when you build the application core image (see :ref:`ug_nrf5340_multi_image` above).

Follow these steps to build and program a multi-image build to the nRF5340 application core and network core:

1. Open Visual Studio code.
#. Click on the :guilabel:`nRF Connect for VS Code` icon in the left navigation bar and then click :guilabel:`Open`.
#. Click on :guilabel:`Add an existing application to workspace...`.
#. In the prompt, navigate to the folder containing the sample you want to build, such as :file:`nrf/samples/bluetooth/peripheral_lbs`.

>    You should now see the selected application in the :guilabel:`Applications` window in the lower left corner.

Reviewer 2: It seems to me that this window does not necessary have to be in the lower left year.
Meybe:
```suggestion
   You should now see the selected application in the :guilabel:`Applications` window of the NRF CONNECT panel.
```

#. Program the application sample using :guilabel:`Target` > :guilabel:`Download zephyr/merged.hex`.

Using |VSC| extension
======================

To build and program separate images with |VSC|, follow the instructions in `Building an application`_.

> To program a multi-image HEX file, you must add the project files for the network core after building, as described in :ref:`ug_nrf5340_ses_multi_image`.

Author: Removed this instruction as this is not applicable for VSC.
